src/server/node_manager.c

```diff
@@ -7128,13 +7128,16 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				} 
 				else {
 					if(presv->ri_qs.ri_state == RESV_UNCONFIRMED) {
-						set_resv_retry(presv, time_now+10);
 						free(phowl);
                 	                	free(execvncopy);
                         	      		return (PBSE_BAD_NODE_STATE);
 					}
-					else
+					else {
 						resv_setResvState(presv, RESV_DEGRADED, RESV_DEGRADED);
+						free(phowl);
+						free(execvncopy);
+						return (PBSE_BAD_NODE_STATE);
+					}
```

Comment:
You can't reject reservations that are not in the unconfirmed state.  If you do then when one occurrence of a standing reservation ends and the next occurrence's nodes are down/stale, the reservation won't get nodes attached to it.  You need to mark the reservation degraded and call set_resv_retry() appropriately.